packages/next-upgrade/index.ts

```diff
@@ -0,0 +1,87 @@
+#!/usr/bin/env node
+
+import prompts from 'prompts'
+import fs from 'fs'
+import { execSync } from 'child_process'
+import path from 'path'
+import { compareVersions } from 'compare-versions'
+
+type Version = 'latest' | 'rc' | 'canary'
+
+interface Response {
+  version: Version
+}
+
+async function run(): Promise<void> {
+  let version: Version
+
+  const specifiedVersion = process.argv[2]?.replace('@', '')
+  if (['latest', 'rc', 'canary'].includes(specifiedVersion)) {
+    // Shortcut
+    version = specifiedVersion as Version
+  } else {
+    const response: Response = await prompts({
+      type: 'select',
+      name: 'version',
+      message: 'What version of Next.js do you want to upgrade to?',
+      choices: [
+        { title: 'Latest Stable', value: 'latest' },
+        { title: 'Latest Release Candidate', value: 'rc' },
+        { title: 'Canary', value: 'canary' },
+      ],
+      initial: 0,
+    })
+    version = response.version
+  }
+
+  // TODO(LichuAcu): support Turborepo
+  const packageJsonPath = path.resolve(process.cwd(), 'package.json')
+  const packageJson = JSON.parse(fs.readFileSync(packageJsonPath, 'utf8'))
+
+  packageJson.dependencies['next'] = version
+
+  try {
+    const res = await fetch(`https://registry.npmjs.org/next/${version}`)
+    const data = await res.json()
+    if (data.peerDependencies) {
+      packageJson.dependencies['react'] = data.peerDependencies['react']
+      packageJson.dependencies['react-dom'] = data.peerDependencies['react-dom']
+    }
+
+    if (data.version) {
+      if (compareVersions(data.version, '15.0.0-canary') >= 0) {
+        const responseTurbopack = await prompts({
+          type: 'confirm',
+          name: 'enable',
+          message:
+            'Turbopack is stable for dev mode in this Next.js version. Do you want to enable it?',
```

Comment:
I find this text too long. Any ideas on a shorter version? @elsigh

Reply:
"Turbopack is now the stable default for dev mode. Enable it?"

Reply:
We also talked about checking for `webpack` custom key/config in next.config and encouraging folks to go prompt v0.dev/chat to rewrite it in the meantime (worked for my raw-loader case)